blockstore/splitstore/splitstore.go

```diff
@@ -435,14 +470,18 @@ func (s *SplitStore) View(cid cid.Cid, cb func([]byte) error) error {
 	//   view can't have its data pointer deleted, which would be catastrophic.
 	//   Note that we can't just RLock for the duration of the view, as this could
 	//    lead to deadlock with recursive views.
-	wg := s.protectView(cid)
-	if wg != nil {
-		defer wg.Done()
-	}
+
+	// What happens if the transaction ends while a view is open? What happens if a transaction
+	// ends then re-opens while a view is open? Highly unlikely... but I wonder if we should
+	// just use the wait-group regardless?
+
```

Comment:
that's fine, we'll just track the references normally.

Reply:
also note, when a view has started before a transaction we will actually defer starting compaction until it's done.

Reply:
I see. My main concern was:

1. Start compaction.
2. Start view.
3. Finish compaction.
4. Start compaction.

From what I could tell, the second compaction wouldn't wait for the view. Is that correct or am I missing something?

Reply:
Ah yes, you are right. Unlikely to happen but possible, so I guess we should just always use the WaitGroup (now that it is persistent).

Reply:
done in https://github.com/filecoin-project/lotus/pull/6474/commits/759594d01c22a63cc920138f7c67285eea3ff051